fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexer.java

```diff
@@ -250,7 +253,17 @@ private IndexingByIndex getIndexerByIndex() {
                             .build();
                     return indexingLauncher(indexingFunc, attemptCount, origPolicy);
                 }
+
                 // No other methods (yet). This line should never be reached.
+                if (LOGGER.isErrorEnabled()) {
+                    LOGGER.error(KeyValueLogMessage.build("Invalid previous indexing type stamp",
+                            LogMessageKeys.CURR_ATTEMPT, attemptCount,
+                            LogMessageKeys.ACTUAL_TYPE, conflictingIndexingTypeStamp)
+                            .addKeysAndValues(common.indexLogMessageKeyValues())
+                            .toString());
+                }
+                common.setIndexStatePrecondition(IndexStatePrecondition.BUILD_IF_DISABLED_REBUILD_IF_WRITE_ONLY);
+                return indexingLauncher(indexingFunc, attemptCount);
```

Comment:
Hmm, I'm trying to understand the logic here, but the basic idea is that if we are trying to continue a previous index build and we don't recognize the index build type, then this will try and rebuild from the beginning?

I'm not sure that we want that, as I think this can only happen if someone makes a logic error in the future and introduces a new indexing method but forgets to update this part of the code. Which is to say, that there's a possibility we get into a weird loop where whenever we resume a build with that new method, we end up losing all progress and starting over.

Reply:
I see. I was trying to address the possibility of a damaged type stamp. But comment makes sense. (`parseTypeStampOrThrow`, in indexingBase, is likely to throw an error in this case). 
Changed to throwing an exception.